### ISSUE TITLE
[Feature] discover-by-docker-labels, match only specific namespace or pattern

### ISSUE THREAD

**Issue by ashleysommer** [mcuadros/ofelia#84]
This is an important feature I'd like to see in the future.
Essentially I'd like to be able to specify a namespace or mask for which the ofelia daemon will match against when discovering jobs via docker-labels.
This is because often a host will have multiple instances of ofelia running within different docker-compose projects.

I'm experiencing a bug in my deployment process right now, where a second docker-compose file, when run, will discover jobs from a completely different docker-compose project which happens to be also running on the host.

I can isolate docker-compose projects using the `-p` flag, like `docker-compose up -p "project-name"` this lets me keep same-name containers from overlapping between docker-compose files. Perhaps it would be good for ofelia to detect if it is running inside a specifc docker-compose project, and restrict the jobs it discovers to only those in the same project.

**Comment by helmut72** [comment]
This would be great. I've tried again Ofelia and it's limitations makes it useless.

It doesn't work running as global container, because it doesn't recognize the labels from containers that will start later. I also use many docker-compose files.

Using an own Ofelia instance for every docker-compose file, the different Ofelia instances recognize labels from Containers which are still running.

**Comment by KevinWhalen** [comment]
A job for [docker-gen](https://github.com/nginx-proxy/docker-gen)? It reads all existing labels, then continuously monitors new containers for labels. It uses those label events to write configuration files from templates.

**Comment by reddec** [comment]
@ashleysommer @KevinWhalen @helmut72 I hit the same problem. Made myself an alternative https://github.com/reddec/compose-scheduler

**Comment by Walter-o** [comment]
I was debugging for a while and eventually figured out the global thing is not supported.

I am a really hardcore performance optimizer so i really wanted to run 1 container with ofelia, 
would also be useful to control all timers in 1 place.

I hope it's possible in the future.

**Comment by Kitof** [comment]
I'm very interested in this feature: ofelia is integrated into pre-configured stacks such as [mailcow](https://github.com/mailcow/mailcow-dockerized), and this interferes with other deployed ofelia, without simple solutions for isolating them.

**Comment by taraspos** [comment]
I'm slowly working to add better test coverage for https://github.com/mcuadros/ofelia/pull/212, and hopefully I will be able to merge it and publish a new release somewhere this weekend.

**Comment by taraspos** [comment]
Just released version [v0.3.9](https://github.com/mcuadros/ofelia/releases/tag/v0.3.9)

Where you can use CLI flags like
```
-f label=com.docker.compose.project=${COMPOSE_PROJECT_NAME}
```
